### PR TITLE
Some category renaming and auto-sorting JSON data 

### PR DESCRIPTION
**Pull request by heoelri**
Summary of changes proposed in this pull request:

- i've re-ordered and renamed some of the categories
- i've added a sort-object to the data processing to auto-order the data based on lens, category, subcategory and type

> IMPORTANT: Make sure you are familiar with the [Contributing](/Contributing.md) guidance. Particularly, **don't edit \*.md files directly - make your changes in the source \*.json files**.

### CODE REVIEW

**Comment by heoelri** [comment]
This is the relevant part here:

https://github.com/Azure/WellArchitected-Assessment/blob/894ffdb659cf7373de516dd4e1afe3226ac9cdea/.github/workflows/generatedocs.yml#L76-L80

It might cause "some" confusion when our github action is re-ordering content in JSON while working on a PR. But it makes the editing experience (modifying json as of now) a bit easier than it is today. As of now the data remains in the individual files - it might make sense to add an additional function to automatically separate the data into individual files like `<lens>-<category>.json` at some point.

**Comment by heoelri** [comment]
Diff is 2 lines - as expected.

image

**Comment by sebader** [comment]
yeah I understand what you did and I guess it makes sense. Hard to really review anything on this PR though ;)

**Comment by heoelri** [comment]
> yeah I understand what you did and I guess it makes sense. Hard to really review anything on this PR though ;)

Yes, the diff lost all its value here.

**Comment by heoelri** [comment]
> > yeah I understand what you did and I guess it makes sense. Hard to really review anything on this PR though ;)
> 
> Yes, the diff lost all its value here.

At least for the JSON files - for the `*.md` files is it still usable.

File: data/categories.json [hunk 1]
```diff
@@ -351,40 +351,40 @@
     "title": "Networking",
     "subCategories": [
```
Comment (sebader) [inline data/categories.json:352]: based on what did you reorder those?

Comment (heoelri) [reply data/categories.json:352]: my gut feeling - starting with some delivery features and services over to networking - including renaming `Loadbalancer` with `Load Balancer`. Feel free to suggest further changes.

**Review by sebader — APPROVED** [review]
Ok, lets have some confidence this didnt break anything :)

**Comment by heoelri** [comment]
There's a git error after merging this into main: 

```console
?? generator/
[main 99727a1] Auto update: Merge pull request #381 from Azure/fix/renaming
 18 files changed, 5963 insertions(+), 5963 deletions(-)
remote: error: GH006: Protected branch update failed for refs/heads/main.        
remote: error: At least 1 approving review is required by reviewers with write access.        
To https://github.com/Azure/WellArchitected-Assessment
 ! [remote rejected] main -> main (protected branch hook declined)
error: failed to push some refs to 'https://github.com/Azure/WellArchitected-Assessment'
```

https://github.com/Azure/WellArchitected-Assessment/runs/3375712795?check_suite_focus=true

@sebader isn't your approval sufficient enough anymore? :)

**Pull request merged**